designs/v1-api.md

```diff
@@ -0,0 +1,558 @@
+default-97v9h   c6gn.large   us-west-2b   ip-192-168-71-87.us-west-2.compute.internal     True    2d7h   spot       default    default
+default-fhzpm   c7gd.large   us-west-2b   ip-192-168-165-122.us-west-2.compute.internal   True    2d7h   spot       default    default
+default-rw4vf   c6gn.large   us-west-2b   ip-192-168-91-38.us-west-2.compute.internal     True    2d7h   spot       default    default
+default-v5qfb   c7gd.large   us-west-2a   ip-192-168-58-94.us-west-2.compute.internal     True    2d7h   spot       default    default
+```
+
+#### Proposed
+
+```
+➜  karpenter git:(main) ✗ kubectl get nodeclaims -A -o wide
+NAME            TYPE         CAPACITY       ZONE         NODE                                            READY   AGE    ID                                      NODEPOOL  NODECLASS
+default-7lh6k   c6gn.large   spot           us-west-2b   ip-192-168-183-234.us-west-2.compute.internal   True    2d7h   aws:///us-west-2b/i-053c6b324e29d2275   default   default
+default-97v9h   c6gn.large   spot           us-west-2b   ip-192-168-71-87.us-west-2.compute.internal     True    2d7h   aws:///us-west-2a/i-053c6b324e29d2275   default   default
+default-fhzpm   c7gd.large   spot           us-west-2b   ip-192-168-165-122.us-west-2.compute.internal   True    2d7h   aws:///us-west-2c/i-053c6b324e29d2275   default   default
+default-rw4vf   c6gn.large   on-demand      us-west-2b   ip-192-168-91-38.us-west-2.compute.internal     True    2d7h   aws:///us-west-2a/i-053c6b324e29d2275   default   default
+default-v5qfb   c7gd.large   spot           us-west-2a   ip-192-168-58-94.us-west-2.compute.internal     True    2d7h   aws:///us-west-2b/i-053c6b324e29d2275   default   default
+```
+
+**Standard Columns**
+
+1. Name
+2. Instance Type
+3. Capacity - Moved from the wide output to the standard output
+4. Zone
+5. Node
+6. Ready
+7. Age
+
+**Wide Columns (-o wide)**
+
+1. ID - Proposing adding Provider ID to make copying and finding instances at the Cloud Provider easier
+2. NodePool Name
+3. NodeClass Name
+
+### Changes to the API
+
+#### NodeClaim Spec Immutability
+
+**Category:** Breaking
+
+Karpenter currently doesn’t enforce immutability on NodeClaims in v1beta1, though we implicitly assume that users should not be acting against these objects after creation, as the NodeClaim lifecycle controller won’t react to any change after the initial instance launch.
+
+Karpenter can make every `spec` field immutable on the NodeClaim after its initial creation. This will be enforced through CEL validation, where you can perform a check like [`self == oldSelf`](https://kubernetes.io/docs/reference/using-api/cel/#language-overview)` to enforce that the fields cannot have changed after the initial apply. Users who are not on K8s 1.25+ that supports CEL will get the same validation enforced by validating webhooks.
+
+## EC2NodeClass API [AWS CloudProvider]
```

Comment:
Should we be merging this here? Should we split out the RFC and open a new one in the provider?

Reply:
I went back and forth on it; however, I thought that coalescing the changes in a single document rather than splitting them out everywhere made more sense. When Azure or other CloudProviders have v1 changes, they are welcome to add these changes to the v1 RFC doc that is included here.

Reply:
Thought about it more: Decided to go ahead and break it out so that the AWS provider repo has its own version